lib/dataset-tools/evaluation/sentence_evaluator.js

```diff
@@ -186,6 +187,7 @@ class SentenceEvaluator {
         const goldFunctions = untypedTargetCode.map((code) => Array.from(getFunctions(code)));
         const goldDevices = untypedTargetCode.map((code) => Array.from(getDevices(code)));
         result.is_primitive = goldFunctions[0].length === 1;
+        result.target_devices = goldDevices;
```

Comment:
goldDevices is only the devices in the target, not in the context, so "cancel" and similar will be accounted to no device at all. Is that the expected behavior?

Reply:
In my head I was including this in the bucket of "utterances without a domain". If I recall correctly you had mentioned these didn't matter much because the immediate goal was to calculate slot accuracy per domain, so I was thinking of lumping these in a "device-agnostic" category. Otherwise, I can keep a "memory" in `CollectSentenceStatistics` that remembers what the last device was, and then assign this device to these utterances. Does this seem better?

Reply:
Yeah probably better to keep them as "no domain" for now, and we can revisit later.